### ISSUE TITLE
transmutation of `getopt`/`argparse`

### ISSUE THREAD

**Issue by jayvdb** [adsharma/py2many#179]
Single file Python scripts is likely to be a use-case which will be mostly functional soon/first, but they usually use `argparse`.   Luckily there is usually a similar library in most languages that we can convert the Python into.  `getopt` is likely to be easier to get working in all languages, and it is worth doing for that reason alone, however it is less used in real Python code this days.  But it Python authors might be willing to downgrade their `argparse` code into `getopt` if it means they can transpile to a larger number of languages.

**Comment by adsharma** [comment]
Yes, argparse -> structopt is a useful initial attempt that could help people writing quick python CLIs and getting a native binary out of it easily.

Having said that I'm concerned about such translation adding a lot of cognitive overhead to pyrs/transpiler.py. We need to figure out a clean interface - some sort of a plugin mechanism for rewriters + dispatch_map.

**Comment by adsharma** [comment]
This is not of much help for existing code, but for new code, I find this more appealing:

https://github.com/mivade/argparse_dataclass/blob/master/tests/example.py

```
from argparse_dataclass import dataclass

@dataclass
class Opt:
    x: int = 42
    y: bool = False
```
